LetsBTF.sol

```diff
@@ -0,0 +1,759 @@
+        uint256 startingTokenId = _nextTokenId();
+        _mint(msg.sender, _quantity);
+
+        uint256 requestId = COORDINATOR.requestRandomWords(
+            chainlinkConfig.keyHash,
+            chainlinkConfig.subId,
+            chainlinkConfig.requestConfirmations,
+            chainlinkConfig.callbackGasLimit,
+            1 // numWords
+        );
+
+        // type casting and storing in a struct saved ~40% gas in tests
+        requestIdToTokenId[requestId] = TokenContainer(uint16(startingTokenId), uint16(_quantity));
+    }
+
+    /* =========== SET RANDOM CREDIT AMOUNT =========== */
+
+    /**
+     * @dev The meat and potatoes. 
+     * This is where the credit is set based off of the random number from Chainlink.
+     */
+    function fulfillRandomWords(uint256 requestId, uint256[] memory randomWords)
+        internal
+        virtual
+        override
+    {
+        TokenContainer memory container = requestIdToTokenId[requestId];
+        delete requestIdToTokenId[requestId];
+
+        uint256 arraySize = uint256(container.quantity);
+        uint256[] memory tokenIds = new uint256[](arraySize);
+        uint256[] memory amounts = new uint256[](arraySize);
+
+        unchecked {
+            // container.quantity will never be greater than 5
+            // tokenId will never be greater than MAX_SUPPLY (10000)
+            for (uint256 i = 0; i < container.quantity; ++i) {
+                uint256 tokenId = container.startId + i;
+
+                uint256 seed = uint256(keccak256(abi.encode(randomWords[0], tokenId))) % 1000;
+                tokenIds[i] = tokenId;
+
+                Credit storage credit = tokenCredits[tokenId];
+                if (credit.totalCredit == 0) {
+                    uint256 cred;
+
+                    if (seed > 0 && seed < 190) {
+                        // 19% chance
+                        cred = 5000; // 50% credit
+                    } else if (seed > 189 && seed < 369) {
+                        // 17.9% chance
+                        cred = 7500; // 75% credit
+                    } else if (seed > 368 && seed < 869) {
+                        // 50% chance
+                        cred = 10000; // 100% credit
+                    } else if (seed > 868 && seed < 969) {
+                        // 10% chance
+                        cred = 12500; // 125% credit
+                    } else if (seed > 968 && seed < 1000) {
+                        // 3% chance
+                        cred = 15000; // 150% credit
+                    } else {
+                        // .10%
+                        cred = 100000; // 10000% credit
+                    }
+
+                    // will never exceed 1E19 (10 ether)
+                    credit.totalCredit = (MINT_PRICE * cred) / 10000;
+                    // Save user gas during first credit claim
+                    // as `claimedCredit` will already be non-zero.
+                    credit.claimedCredit = -1;
```

Comment:
Not necessary.

Changing the `totalCredit` and `claimedCredit` to `uint128` will remove the need to set it to `-1`.